state/block.go

```diff
@@ -0,0 +1,113 @@
+package state
+
+import (
+	"encoding/json"
+	"math/big"
+
+	"github.com/ChainSafe/gossamer/common"
+	"github.com/ChainSafe/gossamer/core/blocktree"
+	"github.com/ChainSafe/gossamer/core/types"
+	"github.com/ChainSafe/gossamer/polkadb"
+)
+
+type blockState struct {
+	bt *blocktree.BlockTree
+	db *polkadb.BlockDB
+}
+
+func NewBlockState() *blockState {
+	return &blockState{
+		bt: &blocktree.BlockTree{},
+		db: &polkadb.BlockDB{},
+	}
+}
+
+var (
+	// Data prefixes
+	headerPrefix    = []byte("hdr") // headerPrefix + hash -> header
+	blockDataPrefix = []byte("hsh") // blockDataPrefix + hash -> blockData
+)
+
+// headerKey = headerPrefix + hash
+func headerKey(hash common.Hash) []byte {
+	return append(headerPrefix, hash.ToBytes()...)
+}
+
+// blockDataKey = blockDataPrefix + hash
+func blockDataKey(hash common.Hash) []byte {
+	return append(blockDataPrefix, hash.ToBytes()...)
+}
+
+func (bs *blockState) GetHeader(hash common.Hash) (types.BlockHeaderWithHash, error) {
+	var result types.BlockHeaderWithHash
+
+	data, err := bs.bt.Db.Db.Get(headerKey(hash))
```

Comment:
Why we are using `bs.bt.Db.Db` for `Get` but `bs.db.Db` for `Put`? If this is how it should be, maybe comments to explain would be helpful here and elsewhere.

Reply:
Good catch! I think this should be `bs.db.Db.Get()`, below is correct.